### DISCUSSION TITLE
WebSocketMessage how duplicate?

### DISCUSSION THREAD

**Discussion by Jobdaneri** [mitmproxy/mitmproxy#7088] [Q&A] [answered]
hello, how duplicate this message?

from mitmproxy import ctx, http, websocket

def websocket_message(flow: websocket.WebSocketMessage):
    if flow.websocket:
        last_message = flow.websocket.messages[-1]
        if last_message.from_client:
            print(f"{last_message.content}")

def start():
    ctx.master.addons.add(websocket_message)

**Comment by Prinzhorn** [comment, marked as answer]
https://github.com/mitmproxy/mitmproxy/blob/main/examples/addons/websocket-inject-message.py
